components/log_wrappers/src/lib.rs

```diff
@@ -61,40 +60,109 @@ impl<T: std::fmt::Debug> slog::Value for DebugValue<T> {
+    Off,
+    On,
+    Marker, // flag is ‹..›
+}
+
+/// RedactOption is exposed to user to manually control the redaction of log
+/// data.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum RedactOption {
+    Flag(bool),
+    Marker,
+}
+
+impl Default for RedactOption {
+    fn default() -> Self {
+        Self::Flag(false)
+    }
+}
+
+impl FromStr for RedactOption {
+    type Err = String;
+    fn from_str(s: &str) -> Result<RedactOption, String> {
+        match s {
+            "" => Ok(RedactOption::default()),
+            "on" | "ON" => Ok(RedactOption::Flag(true)),
+            "off" | "OFF" => Ok(RedactOption::Flag(false)),
+            "marker" | "MARKER" => Ok(RedactOption::Marker),
+            s => Err(format!("expect: marker, on | off, got: {:?}", s)),
+        }
+    }
+}
+
+impl Serialize for RedactOption {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            Self::Flag(flag) => flag.serialize(serializer),
+            Self::Marker => "marker".serialize(serializer),
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for RedactOption {
+    fn deserialize<D>(deseralizer: D) -> Result<RedactOption, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        struct RedactOptionVisitor;
+
+        impl<'de> de::Visitor<'de> for RedactOptionVisitor {
+            type Value = RedactOption;
+
+            fn expecting(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+                fmt.write_str("string or bool")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<RedactOption, E>
+            where
+                E: de::Error,
+            {
+                FromStr::from_str(value).map_err(E::custom)
+            }
+
+            fn visit_bool<E>(self, flag: bool) -> Result<RedactOption, E>
+            where
+                E: de::Error,
+            {
+                Ok(RedactOption::Flag(flag))
+            }
+        }
+
+        deseralizer.deserialize_any(RedactOptionVisitor)
+    }
+}
+
+impl RedactOption {
+    fn convert(&self) -> RedactLevel {
+        match self {
+            Self::Flag(true) => RedactLevel::On,
+            Self::Marker => RedactLevel::Marker,
+            _ => RedactLevel::Off,
+        }
+    }
 }
 
 // Log user data to info log only when this flag is set to false.
-static REDACT_INFO_LOG: AtomicBool = AtomicBool::new(false);
+static REDACT_INFO_LOG: Atomic<RedactLevel> = Atomic::new(RedactLevel::Off);
 
 /// Set whether we should avoid user data to slog.
-pub fn set_redact_info_log(v: bool) {
-    REDACT_INFO_LOG.store(v, Ordering::Relaxed);
-    proto_set_redact_bytes(v);
+pub fn set_redact_info_log(config: RedactOption) {
+    let level = config.convert();
+    REDACT_INFO_LOG.store(level, Ordering::Relaxed);
+    // TODO: make this configuration suitable for protobuf.
+    proto_set_redact_bytes(level == RedactLevel::On || level == RedactLevel::Marker);
```

Comment:
Any plan for adding marker support for rust-protobuf?

Reply:
It's on-going.

> /hold This pr is ready for review.
> 
> But the `marker` mode should also be implemented and supported in `proto-rust` repo. After the implementation in `proto-rust` repo is merged, this pr can be merged.

Reply:
Pls refer to https://github.com/pingcap/rust-protobuf/pull/11. /cc @overvenus

Reply:
All things are done for it.